metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -877,6 +880,39 @@ private void replay(ApiMessage message, Optional<OffsetAndEpoch> snapshotId, lon
         }
     }
 
+    private void checkSnapshotGeneration(long batchSizeInBytes) {
+        newBytesSinceLastSnapshot += batchSizeInBytes;
+        if (newBytesSinceLastSnapshot >= snapshotMaxNewRecordBytes &&
+            snapshotGeneratorManager.generator == null
+        ) {
+            boolean isActiveController = curClaimEpoch != -1;
+            if (!isActiveController) {
+                // The active controller creates in-memory snapshot every time an uncommitted
+                // batch gets appended. The in-active controller can be more efficient and only
+                // create an in-memory snapshot when needed.
+                snapshotRegistry.createSnapshot(lastCommittedOffset);
+            }
+
+            log.info(
+                "Generating a snapshot that includes (epoch={}, offset={}) after {} committed bytes since the last snapshot.",
```

Comment:
Thanks for adding the log message. Spacing seems a bit weird (can we not have "log.info" on its own line?)

Reply:
Not sure if I follow what you are suggesting. I updated this line but let me know if you are suggesting something different.